#!/usr/bin/env python
PACKAGE = "ds_admittance_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("tank_size",    double_t,    0, "the size of the tank", 4,  0, 50)
gen.add("trigger",    double_t,      0, "a threshold (smaller than size that activates the output)", 2,  0, 50)
gen.add("dissipation_rate", double_t, 0, "disspiation rate of the tank",    0.5, 0,   5)
gen.add("force_dead_zone", double_t, 0, "deadzone for force",    .05, 0,   5)
gen.add("torque_dead_zone", double_t, 0, "deadzone for torque",    .05, 0,   5)
gen.add("force_filter_rate", double_t, 0,    "the rate for a first-order low pass filter for input force",    .95, 0,   1)
gen.add("vel_filter_rate", double_t, 0,    "the rate for a first-order low pass filter for output velocity",    .2, 0,   1)

gen.add("acc_linear_max", double_t, 0,    "maximum linear acc",      2, 0,   10)
gen.add("acc_angular_max", double_t, 0,    "maximum angular acc",    .2, 0,   5)
gen.add("vel_linear_max", double_t, 0,    "maximum linear vel",      .3, 0,   1)
gen.add("vel_angular_max", double_t, 0,    "maximum angular vel",    .2, 0,   0.5)



exit(gen.generate(PACKAGE, "ds_admittance_control", "PowerPassFilter"))